protected/controllers/ApiController.php

```diff
@@ -48,71 +49,55 @@ public function actionDump()
 
     public function actionDataset()
 	{
-                $status='Published';
-                $id = Yii::app()->request->getParam('id');
-                $doi= Yii::app()->request->getParam('doi');
-                $result= Yii::app()->request->getParam('result');
-                if(!isset($result))
-                {
-                  $result='all';
-                }
-                if(isset($id))
-                {
-                   try{
-                   $model=  Dataset::model()->findByAttributes(array('id'=>$id,'upload_status'=>$status));}
-                   catch(CDbException $e)
-                   {
+        $status = 'Published';
+        $id = Yii::app()->request->getParam('id');
+        $doi = Yii::app()->request->getParam('doi');
+        $result = Yii::app()->request->getParam('result');
+        $model = null;
+
+        if (!in_array($result, self::RESULTS)) {
+          $result = 'all';
+        }
 
-                            $this->_sendResponse(404,
-                            sprintf('No items where found for dataset id <b>%s</b>',$id) );
-                   }
-                    if(!isset($model))
-                   {
+        try {
+            if ($id) {
+                $model = Dataset::model()->findByAttributes(array('id' => $id, 'upload_status' => $status));
 
-                        $this->_sendResponse(404,
-                            sprintf('No items where found for dataset id <b>%s</b>',$id) );
-                   }
+                if (!$model) {
+                    $this->_sendResponse(404, sprintf('No items where found for dataset id <b>%s</b>',$id));
                 }
-                else{
-                    try{
-                    $model=  Dataset::model()->findByAttributes(array('identifier'=>$doi,'upload_status'=>$status));}
-                    catch(CDbException $e)
-                   {
-
-                            $this->_sendResponse(404,
-                            sprintf('No items where found for dataset doi <b>%s</b>',$doi) );
-                   }
-                   if(!isset($model))
-                   {
-
-                       $this->_sendResponse(404,
-                            sprintf('No items where found for dataset doi <b>%s</b>',$doi) );
-                   }
+            } elseif ($doi) {
+                $model = Dataset::model()->findByAttributes(array('identifier' => $doi, 'upload_status' => $status));
 
+                if (!$model) {
+                    $this->_sendResponse(404, sprintf('No items where found for dataset doi <b>%s</b>',$doi));
                 }
```

Comment:
**issue:** The output message is not clear, for example, what if the doi is 10035, which does not exist in the database, the message makes me feel that 10035 exists but not files or samples associated.
**suggestion:** Could change the message to `The dataset doi 10035 is not found`.
**suggestion:** Also for the id case, which could change to `The dataset $id is not found.`